apps/android/firefox.py

```diff
@@ -90,12 +90,19 @@ def start(self, url="about:blank"):
         ]
 
         # start app
-        device.stop_application(self.APP_NAME)
-        device.launch_activity(
+        self.device.stop_application(self.APP_NAME)
+        self.device.launch_activity(
             self.APP_NAME,
             self.ACTIVITY_NAME,
             extra_args=app_args,
             url=url,
             e10s=True,
             fail_if_running=False,
         )
+
+    def take_screenshot(self, record, path):
+        print("Getting Screenshot")
+        self.device.shell("screencap -p /sdcard/screen.png")
+        sshot_suffix = "record" if record else "replay"
+        self.device.pull("/sdcard/screen.png", "{}_{}.png".format(path, sshot_suffix))
```

Comment:
We should strip the ".mp" from the path. I wonder if `path` should be replaced with a dynamic filename based on the app, url, and mode. We can do this in a follow-up.

Reply:
Yeah, the filename should be as suggestive and short as possible in my opinion.
Currently the filename of the screenshot is the recording filename (which includes the app and test name) followed by the mode. Like: `android-google.mp_(record|replay).png`
You are suggesting something like `(android|firefox|chrome)-(test-name)-(record|replay).png` ?

Reply:
We currently use the same recordings for all apps, but there's a chance this will change in the future. For now, I think `{test_name}.mp` would be suitable for the recording path, and `{test_name}_{mode}_{timestamp}.png` for the screenshots. We can then use the `--path` argument to specify where to write/fine these files.

Reply:
I would actually make a folder `screen_captures` or `screen_captures/test_name` to put the `{test_name}_{mode}_{timestamp}.png`. This way would be a more organized way to find something when getting back.